aks/locals.tf

```diff
@@ -73,14 +73,16 @@ locals {
   storage_config = {
     azure = merge(local.use_managed_identity ? null : {
       account_key = var.logs_storage.storage_account_key
-      }, {
-      use_federated_token = local.use_managed_identity
-      container_name      = var.logs_storage.container
-      account_name        = var.logs_storage.storage_account
-      request_timeout     = "180s"
-      max_retries         = 50
-      min_retry_delay     = "1s"
-      max_retry_delay     = "5s"
+      }, local.use_managed_identity ? {
+      use_federated_token = true
+      } : null,
```

Comment:
Here you have two ternary operators on `local.use_managed_identity` , one where you're using only the `true` condition, and the other where you're using only the `false` condition. So you can simplify it by having only one ternary operator:
```
merge(
  local.use_managed_identity ? {
    use_federated_token = true
  } : {
    account_key = var.logs_storage.storage_account_key
  }
), ...
```

Reply:
I was under the impression that the `true` & `false` results of the ternary expression had to be of the same type (same object in this case). But I just did a test with what you suggested and it passed. I'll rewrite the expression as you suggested, thanks.

Reply:
Ok, so I just did a complete test (with terraform apply) and it didn't pass. Terraform converted the the `true` result of the conditional expression to a `map(string)` while `use_federated_token` expects a boolean. I'm keeping the two ternary operators.

Reply:
Ok, I'm still expecting too much of terraform I guess :). Then I don't have any better syntax to suggest.